### PR TITLE
Fix build warnings

### LINKED ISSUE #1356: Warnings during build
Our builds all include the following warning / diagnostic output:

```
 Reading input file "..\..\..\..\..\Compilers\VisualBasic\Portable\Syntax\Syntax.xml"...
  ..\..\..\..\..\Compilers\VisualBasic\Portable\Syntax\Syntax.xml(3205,8): Invalid input: The element 'node-structure' in namespace 'http://schemas.microsoft.com/VisualStudio/Roslyn/Compiler' has invalid child element 'lm-equiv' in namespace 'http://schemas.microsoft.com/VisualStudio/Roslyn/Compiler'. List of possible elements expected: 'grammar, node-kind, field, child' in namespace 'http://schemas.microsoft.com/VisualStudio/Roslyn/Compiler'.
  Reading input file "..\..\..\..\..\Compilers\VisualBasic\Portable\Syntax\Syntax.xml"...
  291 node structures, 2 enumerations, 639 node kinds
  ..\..\..\..\..\Compilers\VisualBasic\Portable\Syntax\Syntax.xml(3205,8): Invalid input: The element 'node-structure' in namespace 'http://schemas.microsoft.com/VisualStudio/Roslyn/Compiler' has invalid child element 'lm-equiv' in namespace 'http://schemas.microsoft.com/VisualStudio/Roslyn/Compiler'. List of possible elements expected: 'grammar, node-kind, field, child' in namespace 'http://schemas.microsoft.com/VisualStudio/Roslyn/Compiler'.
  291 node structures, 2 enumerations, 639 node kinds
```

This is either an actionable item or the warning should be deleted.  Right now it's just cluttering up the build output and making it harder to track down real errors.

### PR DESCRIPTION
**Pull request by jaredpar**
This fixes a couple of build warnings:

- Bad hint path for MS.VS.Platform.VSEditor.Interop.dll
- Syntax.xml did not conform to the prescribed XML schema

I additionally modified VisualBasicSyntaxGenerator to exit with an error
code in the case the provided XML file did not match the schema.  That
will prevent future regressions on the second warning I fixed.

closes #1356

### CODE REVIEW

**Comment by jaredpar** [comment]
@gafter, @VSadov, @AlekseyTs, @agocke, @theoy

File: src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/XML/ReadTree.vb [hunk 1]
```diff
@@ -15,14 +15,17 @@ Public Module ReadTree
...
 
         Console.WriteLine("Reading input file ""{0}""...", fileName)
 
-        xDoc = GetXDocument(fileName)
-
-        Dim tree As New ParseTree
+        Dim validationError As Boolean = False
+        Dim xDoc = GetXDocument(fileName, validationError)
+        If validationError Then
+            Return False
```
Comment (AlekseyTs) [inline src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/XML/ReadTree.vb]: Set tree to Nothing here or on entry?

File: src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/XML/ReadTree.vb [hunk 2]
```diff
@@ -15,14 +15,17 @@ Public Module ReadTree
     Dim currentFile As String
 
     ' Read an XML file and return the resulting ParseTree object.
-    Public Function ReadTheTree(fileName As String, Optional ByRef xDoc As XDocument = Nothing) As ParseTree
+    Public Function TryReadTheTree(fileName As String, ByRef tree As ParseTree) As Boolean
```
Comment (AlekseyTs) [inline src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/XML/ReadTree.vb]: Consider adding <Out> attribute to make contract of this method more explicit.

Comment (AlekseyTs) [inline src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/XML/ReadTree.vb]: I mean an Out attribute on the 'tree' parameter.

File: src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/XML/ReadTree.vb [hunk 3]
```diff
@@ -59,14 +62,23 @@ Public Module ReadTree
...
 
         tree.FinishedReading()
-        Return tree
+        Return True
     End Function
 
     ' Open the input XML file as an XDocument, using the reading options that we want.
     ' We use a schema to validate the input.
-    Private Function GetXDocument(fileName As String) As XDocument
+    Private Function GetXDocument(fileName As String, ByRef validationError As Boolean) As XDocument
```
Comment (AlekseyTs) [inline src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/XML/ReadTree.vb]: Consider adding an Out attribute for validationError parameter.

**Comment by jaredpar** [comment]
@AlekseyTs thanks for the feedback.  Pushed a new change which addresses your comments.

**Comment by AlekseyTs** [comment]
LGTM

**Comment by tmeschter** [comment]
:+1:

**Pull request closed without merging**